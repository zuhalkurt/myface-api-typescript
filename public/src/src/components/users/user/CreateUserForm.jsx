import React, {useState} from "react";
import { createUser } from "../../../clients/myFaceClients";


export function CreateUserForm() {
    const [name, setName] = useState("");
    const [username, setUsername] = useState("");
    const [email, setEmail] = useState("");
    const [coverImageUrl, setCoverImageUrl] = useState("");
    const [profileImageUrl, setProfileImageUrl] = useState("");

    return <form onSubmit={e => {
        
        e.preventDefault();
        createUser(name, username, email, coverImageUrl, profileImageUrl);
        
    }}>
        <p>Name: {name}, Username:{username}, Email:{email}, CoverImageUrl:{coverImageUrl}, ProfileImageUrl:{profileImageUrl}</p>
        <label for="name">Name:</label>
        <input
            type="text"
            id="name"
            name="name"
            onChange={e=> setName(e.target.value)}        
        />
        <label for="username">Username:</label>
        <input
            type="text"
            id="username"
            name="username"
            onChange={e => setUsername(e.target.value)}
        />
        <label for="email">Email:</label>
        <input
            type="email"
            id="email"
            name="email"
            onChange={e=> setEmail(e.target.value)}        
        />
        <label for="coverImageUrl">CoverImageUrl:</label>
        <input
            type="text"
            id="coverImageUrl"
            name="coverImageUrl"
            onChange={e=> setCoverImageUrl(e.target.value)}        
        />
        <label for="profileImageUrl">ProfileImageUrl:</label>
        <input
            type="text"
            id="profileImageUrl"
            name="profileImageUrl"
            onChange={e=> setProfileImageUrl(e.target.value)}        
        />
        <input type="submit" value="Submit" />
    </form>
}