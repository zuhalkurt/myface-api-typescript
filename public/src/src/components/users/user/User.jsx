import React from "react";
import {Link} from "react-router-dom";
import "./User.scss";

export function User({ user }) {
    return <div className="user">
        <img src={user.profileImageUrl} alt={user.name}/>
        <Link to={`/users/${user.id}`}>
            <h2>{user.name}</h2>
        </Link>
        <p>{user.email}</p>
        <p>{user.username}</p>
        
    </div>
}

