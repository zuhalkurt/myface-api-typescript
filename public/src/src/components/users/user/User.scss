@import "../../../constants.scss";

.user {
    background-color: $darkerBackgroundColour;
    display: grid;
    grid-template-columns: 30vw 1fr;
    grid-template-rows: 0.7fr 0.3fr;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;


    .user-image {
        grid-column: 1 / 2;
        grid-row: 1 / 4;
        width: 30vw;
        height: 30vw;
        
    }

    h2 {
        grid-column: 2 / 3;
        grid-row: 1 /2;
        font-size: $mobile-subtitle-font-size;
        line-height: $mobile-subtitle-line-height;
        margin: 0;
        margin-top: 8vw;
        align-self: center;
    
    }
    .user-email {
        margin: 0;
        grid-column: 2 / 3;
        grid-row: 2 / 3;
    }
    .user-user-name {
        margin: 0;
        grid-column: 2 / 3;
        grid-row: 3 / 4;
    }
}