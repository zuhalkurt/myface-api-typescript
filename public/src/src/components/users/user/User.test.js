import React from 'react';
import { render, screen} from '@testing-library/react';
import { User } from "../user/User";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

test("should match snapshot", async () => {
    const user ={
        name:"name",
        profileImageUrl:"Profile image url",
    }
    const tree = renderer.create(<BrowserRouter> <User user={user}/> </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();})