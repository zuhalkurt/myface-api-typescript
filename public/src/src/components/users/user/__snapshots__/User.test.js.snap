// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  " ",
  <div
    className="user"
  >
    <img
      alt="name"
      className="user-image"
      id="user-profile-image"
      src="Profile image url"
    />
    <a
      href="/users/undefined"
      onClick={[Function]}
    >
      <h2>
        name
      </h2>
    </a>
    <p
      className="user-email"
    />
    <p
      className="user-user-name"
    />
  </div>,
  " ",
]
`;
